hngprojects/hng_boilerplate_nestjs
# feat: mark single notification as read

## What does the PR do?

This feature allows users to mark a specific notification as read. The endpoint takes a notification ID as a parameter and updates the is_read status of the notification.

How should this be manually tested?

1. Login to a user account, and get the access token

2. Make a PATCH request to the `\api\v1\notifications\{notification_id}` endpoint with the following payload:

```json
Authorization: Bearer <access_token>

{
   " is_read": true
}
```

Successful Response
On a successful notification update, the API should return a 200 status code.

```json
{
      "status": "success",
     "status_code": 200,
     "data": {
         "notification_id": "uuid",
         "is_read": true,
        "updated_at": "2024-07-28T12:00:00Z"
      }
}
```

Authentication Error:
When the current user is not properly authenticated to change the organization detail:

```json
{
  "status": "error",
  "message": "Unauthorized request",
  "status_code": 401
}
```

Invalid UserID Error
On Invalid error, if the user is not found with the provided id:
```json
{
  "status": "error",
  "message": "Notification not found",
  "status_code": 404
}
```

Validation:
If any required field is missing or invalid, the response should return a 400 Bad Request status code with appropriate validation error messages.

```json
{
  "status": "error",
  "message": ["notification_id must be a string"],
  "status_code": "400"
}
```
Task

- [x] Implemented PATCH /api/v1/notifications/{notification_id} endpoint in notifications.controller.ts.
- [x] Added method to update the read status of a notification in the NotificationsService.
- [x] Tested endpoint functionality to ensure it handles various scenarios correctly.
- [x] Added unit tests for the NotificationsService.
- [x] Updated API documentation to reflect the new endpoint.

Reference Issue: [Link to Issue](https://linear.app/team-bingo/issue/BAC2-139/clearing-a-single-or-multiple-notification)
